tests/dataset/test_aoi.py

```diff
@@ -81,6 +81,65 @@ def test_bounds_iou(self) -> None:
         expected_iou = 0.013904645827033404
         assert iou == expected_iou
 
+    def test_corrupt_raster(self) -> None:
+        raster_file = "tests/data/massachusetts_buildings_kaggle/22978945_15_uint8_clipped_clipped.tif"
+        with pytest.raises(BaseException):
+            raster = rasterio.open(raster_file)
+
+    def test_image_only(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/sampling_segmentation_binary-multiband_ci.csv")
+        for row in data:
+            aoi = AOI(raster=row['tif'], label=None)
+
+    def test_missing_label(self) -> None:
```

Comment:
I had already developed a test for this. See line 50. I also prefer the implementation above since it doesn't add any data to test data. We can discuss this.